pkg/scanners/cloudformation/parser/parameter.go

```diff
@@ -57,3 +60,67 @@ func (p *Parameter) UpdateDefault(inVal interface{}) {
 		p.inner.Default = passedVal
 	}
 }
+
+type Parameters map[string]any
+
+func (p *Parameters) Merge(other Parameters) {
+	for k, v := range other {
+		(*p)[k] = v
+	}
+}
+
+func (p *Parameters) UnmarshalJSON(data []byte) error {
+	(*p) = make(Parameters)
+
+	if len(data) == 0 {
+		return nil
+	}
+
+	switch {
+	case data[0] == '{' && data[len(data)-1] == '}':
+		// CodePipeline like format
+		var params struct {
+			Params map[string]any `json:"Parameters"`
+		}
+
+		if err := json.Unmarshal(data, &params); err != nil {
+			return err
+		}
+
+		(*p) = params.Params
+	case data[0] == '[' && data[len(data)-1] == ']':
+		{
+			// Original format
+			var params []string
+
+			if err := json.Unmarshal(data, &params); err == nil {
+				for _, param := range params {
+					parts := strings.Split(param, "=")
+					if len(parts) != 2 {
+						return fmt.Errorf("invalid key-value parameter: %q", param)
+					}
+					(*p)[parts[0]] = parts[1]
+				}
+				return nil
+			}
+
+			// CloudFormation like format
+			var cfparams []struct {
+				ParameterKey   string `json:"ParameterKey"`
+				ParameterValue string `json:"ParameterValue"`
+			}
+
+			if err := json.Unmarshal(data, &cfparams); err != nil {
+				return err
+			}
```

Comment:
There's a note in the docs that says:

> Only ParameterKey and ParameterValue are expected keys, command will throw an exception if receives unexpected keys (e.g. UsePreviousValue or ResolvedValue).

I tried with an input such as:
```json
[
    {
        "UsePreviousValue": "Key1",
        "ResolvedValue": "Value1"
    }
]
```

And expectedly I got nothing unmarshaled back, which is what I expected. 

But I'm curious silently missing parsing will end up ignoring some errors down the codepath as the AWS SDK says that it currently throws an exception on such a case. What do you think, should we improve the handling or its fine?

Reply:
I'll add a check